services/worker/src/worker/job_runner.py

```diff
@@ -314,6 +331,10 @@ def should_skip_job(self) -> bool:
             # note: the collection field is named "worker_version" for historical reasons, it might be renamed
             #   "job_runner_version" in the future.
             return False
+        if cached_response["progress"] is not None and cached_response["progress"] < 1.0:
```

Comment:
Could you please add a line in the function's docstring for that case? (or remove the list of cases from the docstring, if you feel it's useless)